light-client/src/supervisor.rs

```diff
@@ -154,6 +157,23 @@ impl Supervisor {
     /// Verify either to the latest block (if `height == None`) or to a given block (if `height == Some(height)`).
     #[pre(self.peers.primary().is_some())]
     fn verify(&mut self, height: Option<Height>) -> VerificationResult {
+        self.verify_inner(height).map(|trusted_lightblock| {
+            // We can now trust this block
+            self.peers
+                .primary_mut()
+                .unwrap() // SAFETY:`verify_inner` wouldn't have returned `Ok` if there were no more primary
```

Comment:
So this can't panic because `verify_inner` would have failed before. Is this (unit)testable?

Reply:
I agree that it's not great, will think about a better and safer way of doing this!